dask/compatibility.py

```diff
@@ -48,6 +52,43 @@ def _getargspec(func):
     def _getargspec(func):
         return inspect.getargspec(func)
 
+    if sys.version_info[1] <= 6:
+        class BZ2File(bz2.BZ2File, BufferedIOBase):
+            def __init__(self, *args, **kwargs):
+                super(BZ2File, self).__init__(*args, **kwargs)
+
+            def readable(self):
+                return 'r' in self.mode
+
+            def writable(self):
+                return 'w' in self.mode
+
+            def seekable(self):
+                return self.readable()
```

Comment:
In general, they are seekable, but it may be inefficient AFAIK. I found a backport of ``BZ2File`` from Python 3 which implements the ``io.BufferedIOBase`` interface: https://github.com/nvawda/bz2file/blob/master/bz2file.py#L153-L157 I'm not sure how to have a better check for ``seekable`` without reimplementing ``BZ2File`` though.

By the way, it may be necessary to use the ``BZ2File`` compatibility implementation in Python 2.7, too. I need to check this again to be sure, but from what I just saw ``BZ2File`` does not implement ``io.BufferedIOBase`` in Python 2.7 either. In that case, I would also add a few more tests, e.g. for ``textblock`` to check that it works with all compression types for all Python versions.